### PR TITLE
Remove Service list usage by RegistryServices in validations

### PR DESCRIPTION
**Pull request by hhovsepy**
RFE https://github.com/kiali/kiali/issues/4528

Replaced ServiceList by RegistryServices in all remaining validations.

For QE:
Full regression testing of validations. https://kiali.io/docs/features/validations/
UI and e2e  automations run.

### CODE REVIEW

**Comment by lucasponce** [comment]
First comment.
Does it need a rebase or any other PR?
I ask as in my first local built I see this regression with the hello-world "bookinfo":

image
image

**Comment by lucasponce** [comment]
More data:
image

**Comment by hhovsepy** [comment]
@lucasponce thank you, let me check.

File: business/istio_validations.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func (in *IstioValidationsService) GetIstioObjectValidations(ctx context.Context
 	go in.fetchRegistryServices(&registryServices, errChan, &wg)
```
Comment (lucasponce) [inline business/istio_validations.go:169]: I see in the business/istio_validations.go that the logic is still fetching both Services and RegistryServices.
Is it something to consolidate later? I've just started to review the code top->down and that comes to my mind.

File: business/istio_validations.go [hunk 2]
```diff
@@ -108,14 +108,14 @@ func (in *IstioValidationsService) GetValidations(ctx context.Context, namespace
 
 func (in *IstioValidationsService) getAllObjectCheckers(namespace string, istioConfigList models.IstioConfigList, exportedResources kubernetes.ExportedResources, services models.ServiceList, workloadsPerNamespace map[string]models.WorkloadList, workloads models.WorkloadList, mtlsDetails kubernetes.MTLSDetails, rbacDetails kubernetes.RBACDetails, namespaces []models.Namespace, registryServices []*kubernetes.RegistryService) []ObjectChecker {
```
Comment (lucasponce) [inline business/istio_validations.go:109]: In this method, perhaps the services are not needed.
If I read the PR correctly.

Comment (hhovsepy) [reply business/istio_validations.go:109]: agree

File: business/istio_validations.go (hunk 1 above)
Comment (hhovsepy) [reply business/istio_validations.go:169]: I would leave that later, as services.Validations are used.

File: business/checkers/authorization/no_host_checker.go [hunk 3]
```diff
@@ -50,9 +49,6 @@ func (n NoHostChecker) validateHost(ruleIdx int, to []*api_security_v1beta.Rule_
 		return nil, true
 	}
 	namespace, clusterName := n.AuthorizationPolicy.Namespace, n.AuthorizationPolicy.ClusterName
```
Comment (lucasponce) [inline business/checkers/authorization/no_host_checker.go:51]: I've found a regression on this one.
Using this example 
https://kiali.io/docs/features/validations/#kia0104---this-host-has-no-matching-entry-in-the-service-registry
the host detection doesn't seem to work well.

File: business/checkers/sidecars/egress_listener_checker.go [hunk 4]
```diff
@@ -92,9 +91,6 @@ func (elc EgressHostChecker) HasMatchingService(host kubernetes.Host, itemNamesp
 	if host.IsWildcard() && host.Namespace == itemNamespace {
```
Comment (lucasponce) [inline business/checkers/sidecars/egress_listener_checker.go:91]: This validation also shows some regression as well (not sure if introduced by this PR or during all this refactoring).
image

**Comment by lucasponce** [comment]
Another regression in the pair VirtualService/DestinationRule:
image
image

**Comment by lucasponce** [comment]
@hhovsepy I've deleted a couple of screenshots, the test was incorrect.
Sorry for the noise.

**Review by lucasponce — CHANGES_REQUESTED** [review]
@hhovsepy mostly ok but I guess the authorization/no_host_checker.go should require another look, just to confirm if the regression exists.

Otherwise, it looks good to me, I've run several tests in my local environment.

File: business/checkers/authorization/no_host_checker.go (hunk 3 above)
Comment (hhovsepy) [reply business/checkers/authorization/no_host_checker.go:51]: the behavior in this PR is the same as in master, can you please provide more details what is wrong here?

all the validations are shown: 
Screenshot from 2022-01-26 17-33-40

Comment (lucasponce) [reply business/checkers/authorization/no_host_checker.go:51]: I don't see the validations in my example:
image

I'll validate if the master branch is working on my side, but my expectation would be to see the same screenshot you've uploaded.

Comment (lucasponce) [reply business/checkers/authorization/no_host_checker.go:51]: In the master branch I have the same screenshot:
image

Comment (lucasponce) [reply business/checkers/authorization/no_host_checker.go:51]: I'm using the "default" namespace, this is the tested configuration:
```
kind: AuthorizationPolicy
apiVersion: security.istio.io/v1beta1
metadata:
  name: httpbin
  namespace: default
  uid: 58a98524-8e7c-4047-9a39-dba7daf0472b
  resourceVersion: '7153'
  generation: 2
  creationTimestamp: '2022-01-27T10:37:16Z'
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: >
      {"apiVersion":"security.istio.io/v1beta1","kind":"AuthorizationPolicy","metadata":{"annotations":{},"name":"httpbin","namespace":"default"},"spec":{"rules":[{"from":[{"source":{"principals":["cluster.local/ns/default/sa/sleep"]}},{"source":{"namespaces":["default"]}}],"to":[{"operation":{"hosts":["wrong","ratings","details.default","reviews.default.svc.cluster.local","productpage.outside","google.com","google.org"],"methods":["GET","/package.service/method"],"paths":["/info*"]}},{"operation":{"methods":["POST"],"paths":["/data"]}}],"when":[{"key":"request.auth.claims[iss]","values":["https://accounts.google.com"]}]}],"selector":{"matchLabels":{"app":"httpbin","version":"v1"}}}}
  managedFields:
    - manager: kubectl-client-side-apply
      operation: Update
      apiVersion: security.istio.io/v1beta1
      time: '2022-01-27T10:37:16Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:kubectl.kubernetes.io/last-applied-configuration': {}
        'f:spec': {}
    - manager: pilot-discovery
      operation: Update
      apiVersion: security.istio.io/v1beta1
      time: '2022-01-27T10:37:17Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          .: {}
          'f:conditions': {}
          'f:observedGeneration': {}
spec:
  rules:
    - from:
        - source:
            principals:
              - cluster.local/ns/default/sa/sleep
        - source:
            namespaces:
              - default
      to:
        - operation:
            hosts:
              - wrong
              - ratings
              - details.default
              - reviews.default.svc.cluster.local
              - productpage.outside
              - google.com
              - google.org
            methods:
              - GET
              - /package.service/method
            paths:
              - /info*
        - operation:
            methods:
              - POST
            paths:
              - /data
      when:
        - key: 'request.auth.claims[iss]'
          values:
            - 'https://accounts.google.com'
status:
  conditions:
    - type: Reconciled
      status: 'True'
      lastProbeTime: '2022-01-27T10:38:58.783262913Z'
      lastTransitionTime: '2022-01-27T10:38:58.783263133Z'
      message: 112/112 proxies up to date.
  observedGeneration: '2'
```

Comment (hhovsepy) [reply business/checkers/authorization/no_host_checker.go:51]: @lucasponce Is the cache enabled on your local env? if it is disabled, registry does not provide any services or configs, so validations will not happen.

Comment (lucasponce) [reply business/checkers/authorization/no_host_checker.go:51]: No, cached is enabled as usual.

Comment (hhovsepy) [reply business/checkers/authorization/no_host_checker.go:51]: tried on new environment, still cannot see the problem

Comment (lucasponce) [reply business/checkers/authorization/no_host_checker.go:51]: Ok, I guess I've found that in my scenario I kept having a VS with a wildcard:
image

And that made that is recognized in the logic:
image

So, I think that's valid. I'll update my env to continue with the review.

Comment (lucasponce) [reply business/checkers/authorization/no_host_checker.go:51]: Ok, that was the case:
image

**Review by lucasponce — APPROVED** [review]

**Comment by lucasponce** [comment]
I didn't find new things. In the end, there was a bad test from my side.
I put it in the QE queue.

**Review by lucasponce — APPROVED** [review]
Code looks but I haven't tested the last rebase.
I'd let QE to run the final testing.

**Comment by prachiyadav** [comment]
@hhovsepy most of the test cases related to Validation is failing in UI regression. Please take a look.
https://master-jenkins-csb-servicemesh.apps.ocp-c1.prod.psi.redhat.com/job/kiali/job/test-jobs/job/kiali-UI-tests-parallel/343/#showFailuresLink

**Comment by hhovsepy** [comment]
> @hhovsepy most of the test cases related to Validation is failing in UI regression. Please take a look. https://master-jenkins-csb-servicemesh.apps.ocp-c1.prod.psi.redhat.com/job/kiali/job/test-jobs/job/kiali-UI-tests-parallel/343/#showFailuresLink

The issue is in inconsistency of the Kiali configmap name between setup testing environment and automation configuration.

**Review by prachiyadav — APPROVED** [review]
Regression test pass successfully

**Pull request merged**